packages/@aws-cdk/aws-autoscaling/README.md

```diff
@@ -42,7 +42,174 @@ selectable by instantiating one of these classes:
+a possible one. You will have to determine what thresholds are right for you).
+
+Note that in order to set up this scaling strategy, you will have to emit a
+metric representing your worker utilization from your instances. After that,
+rou would configure the scaling something like this:
+
+```ts
+const workerUtilizationMetric = new cloudwatch.Metric({
+    namespace: 'MyService',
+    metricName: 'WorkerUtilization'
+});
+
+capacity.scaleOnMetric('ScaleToCPU', {
+  metric: workerUtilizationMetric,
+  scalingSteps: [
+    { upper: 10, change: -1 },
+    { lower: 50, change: +1 },
+    { lower: 70, change: +3 },
+  ],
+
+  // Change this to AdjustmentType.PercentChangeInCapacity to interpret the
+  // 'change' numbers before as percentages instead of capacity counts.
+  adjustmentType: autoscaling.AdjustmentType.ChangeInCapacity,
+});
+```
+
+The AutoScaling construct library will create the required CloudWatch alarms and
+AutoScaling policies for you.
+
+#### Target Tracking Scaling
+
+This type of scaling scales in and out in order to keep a metric around a value
+you prefer. There are four types of predefined metrics you can track, or you can
+choose to track a custom metric. If you do choose to track a custom metric,
+be aware that the metric has to represent instance utilization in some way
+(AutoScaling will scale out if the metric is higher than the target, and scale
+in if the metric is lower than the target).
+
+If you configure multiple target tracking policies, AutoScaling will use the
+one that yields the highest capacity.
+
+The following example scales to keep the CPU usage of your instances around
+50% utilization:
+
+```ts
+autoScalingGroup.scaleOnCpuUtilization('KeepSpareCPU', {
+  targetUtilizationPercent: 50
+});
+```
+
+To scale on average network traffic in and out of your instances:
+
+```ts
+autoScalingGroup.scaleOnIncomingBytes('LimitIngressPerInstance', {
+    targetBytesPerSecond: 10 * 1024 * 1024 // 10 MB/s
+});
+autoScalingGroup.scaleOnOutcomingBytes('LimitEgressPerInstance', {
+    targetBytesPerSecond: 10 * 1024 * 1024 // 10 MB/s
+});
+```
+
+To scale on the average request count per instance (only works for
+AutoScalingGroups that have been attached to Application Load
+Balancers):
+
+```ts
+autoScalingGroup.scaleOnRequestCount('LimitRPS', {
+    targetRequestsPerSecond: 1000
+});
+```
+
+#### Scheduled Scaling
+
+This type of scaling is used to change capacities based on time. It works by
+changing `minCapacity`, `maxCapacity` and `desiredCapacity` of the
+AutoScalingGroup, and so can be used for two purposes:
+
+* Scale in and out on a schedule by setting the `minCapacity` high or
+  the `maxCapacity` low.
+* Still allow the regular scaling actions to do their job, but restrict
+  the range they can scale over (by setting both `minCapacity` and
+  `maxCapacity` but changing their range over time).
+
+A scheduled is expressed as a cron expression. There is a `Cron` helper class
```

Comment:
Cron: can we extract it to a common module so we can reuse it in CloudWatch Events? [not critical for this PR of course, but at least raise an issue]

Reply:
Maybe. Two points against this: 

* There are a couple of Cron syntaxes floating round. For example, for application autoscaling and EC2 autoscaling, the two syntaxes look like this:

```
cron(0 18 * * ?)
0 18 * * *
```

Note the `cron()` function, and the difference between `* and ?` for the DOW field.

Also, I think it's convenient that you use the `Cron` class from the library X if you're creating cron expression for library X. It would be less convenient if you had to find library Y to use its `Cron` class.

Reply:
We could create a shared cron library and expose the relevant instance (perm cron-syntax) from the various usage points